Tasks/Common/dotNetCliWrapper.ts

```diff
@@ -112,27 +111,94 @@ export class DotNetCliWrapper {
 
         for (const arg of args) {
             dotnet.arg(arg)
         }
 
         dotnet.line(additionalArgs)
 
-        const execOptions = {
+        const execOptions: any = {
             cwd: this.cwd,
             env: this.env
         }
 
+        if (additionalCommandLineOptions) {
+            // tslint:disable-next-line: no-unsafe-any
+            for (const key of Object.keys(additionalCommandLineOptions)) {
+                // tslint:disable-next-line: no-unsafe-any
+                execOptions.key = additionalCommandLineOptions[key]
+            }
+        }
+
         // tslint:disable-next-line: no-unsafe-any
-        await dotnet.exec(execOptions as any)
+        return await dotnet.exec(execOptions)
+    }
+
+    private async restore(): Promise<number> {
+        return await this.execute(['restore'], '')
+    }
+
+    private async checkForGlobalLambdaToolsInstalled(): Promise<boolean> {
+        try {
+            const returnCode = await this.execute(['lambda', 'help'], '', { silent: true })
+            if (returnCode !== 0) {
+                return false
+            }
+        } catch (exception) {
+            return false
+        }
+
+        return true
+    }
+
+    private async installGlobalTools(): Promise<boolean> {
+        try {
+            const returnCode = await this.execute(['tool', 'install', '-g', 'Amazon.Lambda.Tools'], '')
+            if (returnCode === 0) {
+                return true
+            }
+        } catch (e) {}
+
+        // If something went wrong in the last step, we try to update the tools instead
+        try {
+            const returnCode = await this.execute(['tool', 'update', '-g', 'Amazon.Lambda.Tools'], '')
+            if (returnCode === 0) {
+                return true
+            }
+        } catch (e2) {
+            return false
+        }
+
+        return false
+    }
+
+    public static async buildDotNetCliWrapper(cwd: string, env: any, dotnetCliPath: string): Promise<DotNetCliWrapper> {
+        const wrapper = new DotNetCliWrapper(cwd, env, dotnetCliPath)
+        tl.debug(tl.loc('StartingDotNetRestore'))
+        // Restore, this will install the lambda cli if it's < version 3
+        await wrapper.restore()
+        if (await wrapper.checkForGlobalLambdaToolsInstalled()) {
+            return wrapper
+        }
+
+        tl.debug(tl.loc('InstallingOrUpdatingLambdaTools'))
+        if (!(await wrapper.installGlobalTools())) {
+            // if checking for lambda tools fails and installing them fails, we are probably on the
+            // wrong instance type because we were unable to install
+            throw new Error(
+                'Unable to install Amazon.Lambda.Tools! Are you using the correct hosted ' +
+                    "agent type? Refer to Microsoft's guide for the correct hosted agent for .NET Core" +
+                    'to make sure: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted? ' +
```

Comment:
There's a space after the ? that should probably be removed.

Reply:
Fixed